server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -449,11 +449,11 @@ public void onFailure(Exception e) {
      * or return a new DocumentSizeObserver that will be used when parsing.
      */
     private static DocumentSizeObserver getDocumentSizeObserver(DocumentParsingProvider documentParsingProvider, IndexRequest request) {
-        if (request.getNormalisedBytesParsed() != -1) {
+        if (request.getNormalisedBytesParsed() > 0) {
             return documentParsingProvider.newFixedSizeDocumentObserver(request.getNormalisedBytesParsed());
         } else if (request.getNormalisedBytesParsed() == 0) {
             return DocumentSizeObserver.EMPTY_INSTANCE;
-        }
+        } // request.getNormalisedBytesParsed() == -1, meaning an IndexRequest has a default value (not being used in updates or ingest)
```

Comment:
I don't think I understand the comment 😳 
I though depending on how the data arrives, we might parse it early (so it's > 0 at this point). Otherwise we yet have to parse the data going forward? Or am I confusing things?

Reply:
you are correct.
-1 means that the IndexRequest was created parsing has not happened yet.
perhaps `-1, meaning an IndexRequest has a default value, parsing has not happened yet` ?

Reply:
👍 much better, maybe even clearer as follows:
```-1, meaning normalisedBytesParsed isn't set as parsing wasn't done yet```